pkg/chunk/composite_store.go

```diff
@@ -174,6 +175,22 @@ func (c compositeStore) GetChunkRefs(ctx context.Context, userID string, from, t
 	return chunkIDs, fetchers, err
 }
 
+func (c compositeStore) GetChunkFetcher(chunk Chunk) *Fetcher {
```

Comment:
Can we only pass single timestamp instead of `Chunk`?

(It is also not clear what should be returned if chunk crossed the boundary between two schemas)

Reply:
It sounds good, I wanted to do it like that but the code would change based on whether you are sending start time of chunk or end time of chunk. We will have to make it clearer by renaming the function or something that you are supposed to send end time of chunk otherwise the caller could do a mistake by sending chunk start time instead of the end time.

What do you think?

Reply:
> It sounds good, I wanted to do it like that but the code would change based on whether you are sending start time of chunk or end time of chunk. We will have to make it clearer by renaming the function or something that you are supposed to send end time of chunk otherwise the caller could do a mistake by sending chunk start time instead of the end time. Maybe a comment and a relevant function parameter name like `chunkEndTime` should be enough?
> 
> What do you think?

I think the reason to use timestamp as parameter instead of full chunk would be to remove this question from the table *in the method*. There should still be some consistency between the calls, but now it would be up to client to be consistent about sending start or end timestamps.

I am wondering what difference does it really make whether we pass start or end time to the method though -- In both cases, the fetcher must be able to fetch the chunk, right?

Reply:
As discussed, I have added validation for not sharing the same date with multiple periodic configs.